typescript/cli/src/deploy/utils.ts

```diff
@@ -86,12 +86,30 @@ export async function runDeployPlanStep({
   );
 
   if (skipConfirmation) return;
+  await confirmExistingMailbox(context, chain);
   const isConfirmed = await confirm({
     message: 'Is this deployment plan correct?',
   });
   if (!isConfirmed) throw new Error('Deployment cancelled');
 }
 
+async function confirmExistingMailbox(
+  context: CommandContext,
+  chain: ChainName,
+) {
+  const addresses = await context.registry.getChainAddresses(chain);
+  if (addresses?.mailbox) {
+    const isConfirmed = await confirm({
+      message: `Mailbox already exists at ${addresses.mailbox}. Are you sure you want to deploy a new mailbox and overwrite existing registry artifacts?`,
+      default: false,
+    });
+
+    if (!isConfirmed) {
+      throw Error('Deployment cancelled');
```

Comment:
I find this slightly odd behavior, answering `no` is an error?

Reply:
not sure if `detectAndConfirmOrPrompt` is useful here

Reply:
Alternatives would be to return some value here and check if in callers or just `process.exit(1)` if we def want to end execution here